### ISSUE TITLE
Mysql backend provision error

### ISSUE THREAD

**Issue by MyTheValentinus** [LibrIT/passhport#398]
**Describe the bug**
I just would change my backend to mysql, i follow your documention, create DB, create User etc..
But i launch the provision script `/home/passhport/passhport-run-env/bin/python /home/passhport/passhport/passhportd/db_create.py` he return a error on `password` column.

MariaDB Version: 10.3.17

```
pymysql.err.InternalError: (1170, "BLOB/TEXT column 'password' used in key specification without a key length")



Traceback (most recent call last):
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1244, in _execute_context
    cursor, statement, parameters, context
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/cursors.py", line 170, in execute
    result = self._query(query)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/cursors.py", line 328, in _query
    conn.query(q)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/connections.py", line 517, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/connections.py", line 732, in _read_query_result
    result.read()
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/connections.py", line 1075, in read
    first_packet = self.connection._read_packet()
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/connections.py", line 684, in _read_packet
    packet.check_error()
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/protocol.py", line 220, in check_error
    err.raise_mysql_exception(self._data)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/err.py", line 109, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.InternalError: (1170, "BLOB/TEXT column 'password' used in key specification without a key length")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/passhport/passhport/passhportd/db_create.py", line 19, in <module>
    db.create_all()
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/flask_sqlalchemy/__init__.py", line 1033, in create_all
    self._execute_for_all_tables(app, bind, 'create_all')
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/flask_sqlalchemy/__init__.py", line 1025, in _execute_for_all_tables
    op(bind=self.get_engine(app, bind), **extra)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/sql/schema.py", line 4294, in create_all
    ddl.SchemaGenerator, self, checkfirst=checkfirst, tables=tables
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2036, in _run_visitor
    conn._run_visitor(visitorcallable, element, **kwargs)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1607, in _run_visitor
    visitorcallable(self.dialect, self, **kwargs).traverse_single(element)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/sql/visitors.py", line 132, in traverse_single
    return meth(obj, **kw)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/sql/ddl.py", line 781, in visit_metadata
    _is_metadata_operation=True,
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/sql/visitors.py", line 132, in traverse_single
    return meth(obj, **kw)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/sql/ddl.py", line 833, in visit_table
    self.traverse_single(index)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/sql/visitors.py", line 132, in traverse_single
    return meth(obj, **kw)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/sql/ddl.py", line 862, in visit_index
    self.connection.execute(CreateIndex(index))
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 988, in execute
    return meth(self, multiparams, params)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/sql/ddl.py", line 72, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1050, in _execute_ddl
    compiled,
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1248, in _execute_context
    e, statement, parameters, cursor, context
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1466, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 152, in reraise
    raise value.with_traceback(tb)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1244, in _execute_context
    cursor, statement, parameters, context
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/cursors.py", line 170, in execute
    result = self._query(query)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/cursors.py", line 328, in _query
    conn.query(q)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/connections.py", line 517, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/connections.py", line 732, in _read_query_result
    result.read()
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/connections.py", line 1075, in read
    first_packet = self.connection._read_packet()
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/connections.py", line 684, in _read_packet
    packet.check_error()
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/protocol.py", line 220, in check_error
    err.raise_mysql_exception(self._data)
  File "/home/passhport/passhport-run-env/lib/python3.7/site-packages/pymysql/err.py", line 109, in raise_mysql_exception
    raise errorclass(errno, errval)
sqlalchemy.exc.InternalError: (pymysql.err.InternalError) (1170, "BLOB/TEXT column 'password' used in key specification without a key length")
[SQL: CREATE INDEX ix_passentry_password ON passentry (password)]
(Background on this error at: http://sqlalche.me/e/2j85)

```
Thanks you ;) !

**Comment by MyTheValentinus** [comment]
Second run:

```
sqlalchemy.exc.InternalError: (pymysql.err.InternalError) (1071, 'Specified key was too long; max key length is 3072 bytes')
[SQL: 
CREATE TABLE user (
	id INTEGER NOT NULL AUTO_INCREMENT, 
	name VARCHAR(120) NOT NULL, 
	sshkey VARCHAR(5000) NOT NULL, 
	sshkeyhash VARCHAR(5000), 
	comment VARCHAR(5000), 
	superadmin BOOL, 
	logfilesize VARCHAR(30), 
	PRIMARY KEY (id), 
	UNIQUE (sshkey), 
	CHECK (superadmin IN (0, 1))
)

]
(Background on this error at: http://sqlalche.me/e/2j85)
```

**Comment by elg** [comment]
To be honnest we are working a lot to maintain postgresql on master branch. We plan to do the updates on MySQL on major versions.
I'll still give it a try in next days : this issue seems trivial to fix, but i'll have some tests to run.

**Comment by elg** [comment]
It's a limitation on MySQL: you can't index a field larger than 3k or something. I'll try to workaround this limitation.

**Comment by elg** [comment]
This issue is now resolved with several commits.
FYI: the sshkey is not considered unique in database cause it could limit us in various way if we want to support mysql. We are using SHA256 hash as unique and even a highly unlikely collision would only force a sshkey regeneration by a user.
I reworked the user management (creation/edition) to handle all of this and improve code performances. This can lead to some unexpected bugs (but I hope not :) )
